docs/arcadia-users-group/20231031-intro-to-r/lesson.md

```diff
@@ -0,0 +1,1674 @@
+## # ℹ 3 variables: species_id <chr>, plot_id <dbl>, mean_weight <dbl>
+```
+
+We get back 0 rows. There is no `mean_weight` for the species `BA` in plot `1`. This either happened because no `BA` were ever caught in plot `1`, or because every `BA` caught in plot `1` had an `NA` weight value and all the rows got removed when we used `filter(!is.na(weight))` in the process of making `sp_by_plot`. Because there are no rows with that species + plot combination, in our pivoted data.frame, the value gets filled with `NA`.
+
+There is another `pivot_` function that does the opposite, moving data from a wide to long format, called `pivot_longer()`. It takes 3 arguments: `cols` for the columns you want to pivot, `names_to` for the name of the new column which will contain the old column names, and `values_to` for the name of the new column which will contain the old values.
+
+![](fig/pivot_longer.png){alt='Diagram depicting the behavior of `pivot_longer()` on a small tabular dataset.'}
+
+We can pivot our new wide data.frame to a long format using `pivot_longer()`. We want to pivot all the columns except `species_id`, and we will use `PLOT` for the new column of plot IDs, and `MEAN_WT` for the new column of mean weight values.
+
+
+```r
+sp_by_plot_wide %>% 
+  pivot_longer(cols = -species_id, names_to = "PLOT", values_to = "MEAN_WT")
+```
+
+```
+## # A tibble: 432 × 3
+## # Groups:   species_id [18]
+##    species_id PLOT  MEAN_WT
+##    <chr>      <chr>   <dbl>
+##  1 BA         3         8  
+##  2 BA         21        6.5
+##  3 BA         1        NA  
+##  4 BA         2        NA  
+##  5 BA         4        NA  
+##  6 BA         5        NA  
+##  7 BA         6        NA  
+##  8 BA         7        NA  
+##  9 BA         8        NA  
+## 10 BA         9        NA  
+## # ℹ 422 more rows
+```
+
+One thing you will notice is that all those `NA` values that got generated when we pivoted wider. However, we can filter those out, which gets us back to the same data as `sp_by_plot`, before we pivoted it wider.
+
+
+```r
+sp_by_plot_wide %>% 
+  pivot_longer(cols = -species_id, names_to = "PLOT", values_to = "MEAN_WT") %>% 
+  filter(!is.na(MEAN_WT))
+```
+
+```
+## # A tibble: 300 × 3
+## # Groups:   species_id [18]
+##    species_id PLOT  MEAN_WT
+##    <chr>      <chr>   <dbl>
+##  1 BA         3         8  
+##  2 BA         21        6.5
+##  3 DM         3        41.2
+##  4 DM         21       41.5
+##  5 DM         1        42.7
+##  6 DM         2        42.6
+##  7 DM         4        41.9
+##  8 DM         5        42.6
+##  9 DM         6        42.1
+## 10 DM         7        43.2
+## # ℹ 290 more rows
+```
+
+Data are often recorded in spreadsheets in a wider format, but lots of `tidyverse` tools, especially `ggplot2`, like data in a longer format, so `pivot_longer()` is often very useful.
+
+### Joining data together with `dplyr`
+
+Another operation that is often useful is to join two data.frames together based on shared columns.
```

Comment:
I would maybe quickly demo this, can cut it out at the end if running out of time (you already have quite a bit of material!) but getting to this point would equip learners with most of what they need. Joins are usually what people need for a majority of things

Reply:
added!